hw/mcu/dialog/da1469x/src/arch/cortex_m33/da1469x_m33_sleep.S

```diff
@@ -149,6 +166,32 @@ da1469x_m33_sleep:
     orr     r1, r1, #0x1000
     str     r1, [r2, #0]
 
+/*
+ * Flash is likely in QPI mode but QSPIC assumes it is in single mode so we
+ * should make sure both are in sync by forcing exit from QPI mode and then
+ * re-enter it. We need to do this before we start executing code from flash
+ * again.
+ */
+    ldr     r0, =QSPIC_BASE
+    ldr     r1, [r0, #(QSPIC_CTRLMOD_OFFSET)]
+    bic     r1, r1, #1      /* Clear automode */
+    orr     r1, r1, #0x3c   /* Set IO2/IO3 DAT and OEN */
+    str     r1, [r0, #(QSPIC_CTRLMOD_OFFSET)]
+    mov     r1, #1          /* Set single */
+    str     r1, [r0, #(QSPIC_CTRLBUS_OFFSET)]
+    mov     r1, #8          /* Enable CS */
+    str     r1, [r0, #(QSPIC_CTRLBUS_OFFSET)]
+    mov     r1, #0xff       /* Exit QPI mode */
+    strb    r1, [r0, #(QSPIC_WRITEDATA_OFFSET)]
+    mov     r1, #16         /* Disable CS */
+    str     r1, [r0, #(QSPIC_CTRLBUS_OFFSET)]
```

Comment:
code above this line is not necessary since we are already in single mode - it was done before wfi